dsgrid/utils/dataset.py

```diff
@@ -254,8 +255,8 @@ def handle_dimension_association_errors(
     dataset_table: DataFrame,
     dataset_id: str,
 ) -> None:
-    """Record missing dimension record combinations in a CSV file and log an error."""
-    out_file = f"{dataset_id}__missing_dimension_record_combinations.csv"
+    """Record missing dimension record combinations in a Parquet file and log an error."""
+    out_file = Path(f"{dataset_id}__missing_dimension_record_combinations.parquet")
```

Comment:
So we are still generating all missing combinations and writing that out? I thought that was where the slowdown was ... What am I missing?